### PR TITLE
Move computeAt logic to separate file/class.

### PR DESCRIPTION
**Pull request by csarofeen**
Cleanup before re-writing computeAt.

### CODE REVIEW

**Review by naoyam — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/compute_at.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include <c10/util/Exception.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+
+struct TensorView;
+
+struct ComputeAt {
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/compute_at.h:16]: can we make this a `class` (and order the members from public, then private methods and finally private data) ?

or even better - split the configuration from the action and have a simple interface like:

```
struct ComputeAtContext {
  TensorView* producer = nullptr;
  TensorView* consumer = nullptr;
  unsigned int consumer_position = 0;
};

void applyComputeAt(const ComputeAtContext& context);
```

This way everything irrelevant to the users are really hidden and we have a simple & small interface. It also have a few other small benefits:

1. Encapsulating configuration as a struct avoids the problem of "many arguments" 
    (https://abseil.io/tips/173 has a decent explanation)

2. Calling a function seems more intuitive than applying an action as a side effect of creating an object

File: torch/csrc/jit/codegen/cuda/compute_at.cpp [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+    TensorView* consumer,
+    int producer_compute_at_axis) {
+  // Reset view otherwise will conflict with replay. Don't think this is true
+  // anymore.
+  producer->clearComputeAt();
+  TransformReplay::replayCasP(consumer, producer, producer_compute_at_axis);
+  producer->setComputeAt(consumer, producer_compute_at_axis);
+}
+
+namespace {
```
Comment (tlemo) [inline torch/csrc/jit/codegen/cuda/compute_at.cpp:55]: prefer static functions to anonymous namespaces?

**Review by tlemo — COMMENTED** [review]
This is great! 

I added a suggestion for a simpler / thinner inteface.

**Comment by csarofeen** [comment]
closed as part of: https://github.com/csarofeen/pytorch/pull/132

**Pull request closed without merging**